library/platform_util.c

```diff
@@ -20,13 +20,22 @@
  *  This file is part of Mbed TLS (https://tls.mbed.org)
  */
 
+/*
+ * Ensure gmtime_r is available even with -std=c99; must be included before
+ * config.h, which pulls in glibc's features.h. Harmless on other platforms.
+ */
+#if !defined(_POSIX_C_SOURCE)
+#define _POSIX_C_SOURCE 200112L
+#endif
+
 #if !defined(MBEDTLS_CONFIG_FILE)
 #include "mbedtls/config.h"
 #else
 #include MBEDTLS_CONFIG_FILE
 #endif
 
 #include "mbedtls/platform_util.h"
+#include "mbedtls/threading.h"
```

Comment:
Shouldn't this be surrounded by #if defined(MBEDTLS_THREADING_C)?

Reply:
We could, but then we should also guard it by the other flags determining whether we call back to our default `mbedtls_platform_gmtime_r()` implementation.